content/200-concepts/100-components/02-prisma-client/260-client-extensions/130-result.mdx

```diff
@@ -0,0 +1,152 @@
+tocDepth: 3
+---
+
+<TopBlock>
+
+You can use the `result` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component type to add custom fields and methods to query results. We introduced this feature in version 4.7.0.
+
+</TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add custom fields or methods to query results
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an _extended client_. An extended client is a variant of the standard Prisma Client that is wrapped by one or more extensions.
+
+Use the `result` extension component to add custom fields and methods to query results.
+
+To add a custom [field](#add-a-custom-field-to-query-results) or [method](#add-a-custom-method-to-the-result-object) to query results, use the following structure. In this example, we add a custom field to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  result?: {
+    user: {                   // in this case, we extend the `user` model
+      myComputedField: {      // the name of the new computed field
+        needs: { ... }
+        compute(): { ... }
+      },
+    },
+  },
+});
+```
+
+The parameters are as follows:
+
+- `name`: (optional) specifies a name for the extension that appears in error logs.
+- `result`: defines new fields and methods to the query results.
+- `needs`: an object which describes the dependencies of the result field.
+- `compute`: an object that contains methods that define how the virtual field is computed when it is accessed. You must declare the same properities in both `compute` and `needs`.
+
+### Add a custom field to query results
+
+You can use the `result` extension component to add fields to query results. These fields are computed at runtime and are type-safe.
+
+In the following example, we add a new virtual field called `fullName` to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  result: {
+    user: {
+      fullName: {
+        needs: { firstName: true, lastName: true }, // the dependencies
+        compute(user) {
+          // the computation logic
+          return `${user.firstName} ${user.lastName}`
+        },
+      },
+    },
+  },
+})
+
+const user = await xprisma.user.findFirst()
+
+// return the user's full name, such as "John Doe"
+console.log(user.fullName)
+```
+
+In above example, the input `user` is automatically typed according to the object defined in `needs`. `firstName` and `lastName` are of type `string`, because they are specified in `needs`. If they are not specified in `needs`, then their type is `string | undefined`.
```

Comment:
```suggestion
In above example, the input `user` of `compute` is automatically typed according to the object defined in `needs`. `firstName` and `lastName` are of type `string`, because they are specified in `needs`. If they are not specified in `needs`, then they cannot be accessed.
```